### PR TITLE
add test for list rule with service group id

### PR DESCRIPTION
**Pull request by cam-d12**
Signed-off-by: Cameron Davis <cjdavis@ibm.com>

## PR summary

This PR adds CBR support for the new service_group_id parameter in rules.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

No current behavior should be modified. Adds the further control to be able to apply rules to service groups

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: modules/context-based-restrictions/src/test/java/com/ibm/cloud/platform_services/context_based_restrictions/v1/ContextBasedRestrictionsIT.java [hunk 1]
```diff
@@ -614,6 +629,86 @@ public void testListRules() throws Exception {
...
+            // Invoke operation
+            Response<RuleList> listResponse = service.listRules(listRulesOptions).execute();
+            // Validate response
+            assertNotNull(listResponse);
+            assertEquals(listResponse.getStatusCode(), 200);
+
+            RuleList ruleListResult = listResponse.getResult();
+
+            assertNotNull(ruleListResult);
+            assertEquals(ruleListResult.getTotalCount(), 1)
```
Comment (soohoowoohoo) [inline modules/context-based-restrictions/src/test/java/com/ibm/cloud/platform_services/context_based_restrictions/v1/ContextBasedRestrictionsIT.java:694]: can we assert the rule ID from line 674 is the rule that is returned?

**Review by soohoowoohoo — CHANGES_REQUESTED** [review]

**Review by soohoowoohoo — COMMENTED** [review]
LGTM

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**